### ISSUE TITLE
APSFOUND-54: Improvements to variable resolution for docker scenarios

### ISSUE THREAD

**Issue by brenthdaniel** [OpenLiberty/open-liberty#6627]
This feature proposes several changes to variable resolution to better support docker:

Match liberty variables to environment variables without env. prefix.
Match liberty variables to environment variables replacing non-alphanum with _
Match liberty variables to environment variables replacing non-alphanum with _ and changing to upper case
Allow default values for variables which would allow you to define a default, but override using environment variables, good for docker.
Design issues for these:

https://github.com/OpenLiberty/open-liberty/issues/6209

https://github.com/OpenLiberty/open-liberty/issues/6186

**Comment by atosak** [comment]
https://bigblue.aha.io/features/APSFOUND-54

**Comment by marikaj123** [comment]
@brenthdaniel - Do you have English files that require translation?

**Comment by brenthdaniel** [comment]
@marikaj123 Only the ones in the previous PR #6623 that was merged on 2/28

**Comment by marikaj123** [comment]
@brenthdaniel - As long as you merged them by March 1 then they will be in the returned translated files.

**Comment by skasund** [comment]
L2 has requested STE slides for this feature.  The STE template can be found at the link below: 

https://ibm.box.com/s/1an42g7zdgmaj84w7dft0indqfgi8ffm

Please upload the completed slides to the same STE Archive BOX folder.  Thanks!

**Comment by brenthdaniel** [comment]
@skasund I uploaded the slides to the archive.

**Comment by skasund** [comment]
Thank you for the slides.  I've provided the STE approval.

**Comment by atosak** [comment]
ID issue: https://github.ibm.com/was-liberty/liberty-docs/issues/1133

**Comment by marikaj123** [comment]
@brenthdaniel - Please confirm the translated files for this feature are in Master.

**Comment by sabolo** [comment]
Please can you complete the feature test summary so that the FAT approval can proceed?  I have added the child issue for you and inserted the template.

**Comment by donbourne** [comment]
**Serviceability Approval Comment** - Please answer the following questions for serviceability approval:

1.  WAD -- does the WAD identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR?  Have these issues been addressed in the implementation?

2.  Test and Demo -- As part of the serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. L2, test team, or another development team).  
a) What problem paths were tested and demonstrated? 
b) Who did you demo to?
c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that L2 should be able to quickly address those problems without need to engage L3?

3.  SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them.  Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.
a) Who conducted SVT tests for this feature?
b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3?  

4.  Which L2 / L3 queues will handle PMRs for this feature?  Ensure they are present in the contact reference file and in the queue contact summary, and that the respective L2/L3 teams know they are supporting it.  Ask Don Bourne if you need links or more info.

**Comment by gscottj** [comment]
Based on the description, this feature is not introducing any new end user user interfaces which require accessibility testing.

**Comment by chirp1** [comment]
- Alasdair is going to do an update in the open liberty config overview asciidoc.
- ID will add a link from the https://www.ibm.com/support/knowledgecenter/en/SSEQTP_liberty/com.ibm.websphere.wlp.doc/ae/cwlp_config.html to the OL config overview assciidoc.

**Comment by marikaj123** [comment]
@brenthdaniel - Please provide us with a status on this feature?

**Comment by brenthdaniel** [comment]
@marikaj123 Yes, the translated files are in master

@sabolo The feature test summary has been updated

**Comment by brenthdaniel** [comment]
@donbourne 

1) The WAD lists serviceability as N/A because of the limited nature of the changes. 
2) a) We demoed a variable definition that is missing a value and defaultValue, as well as a variable definition that contains a cycle. 
b) Several members of the kernel team
c) Yes, though it was brought up that a couple of lines of trace could be added for L3 purposes.
3) SVT believes that no new testing is needed for this feature, though it may be exercised in future scenarios.
4) LL3KRN,103 / WAS L3: Liberty Kernel

**Comment by dave-waddling** [comment]
@brenthdaniel Thanks for completing the Feature Test Summary. I've finished the review and have added the FAT Focal Approval.
